tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/text/ByteBufLineReader.java

```diff
@@ -197,6 +197,6 @@ public ByteBuf readLineBuf(AtomicInteger reads) throws IOException {
       }
     }
     reads.set(readBytes);
-    return buffer.slice(startIndex, readBytes - newlineLength);
+    return buffer.slice(startIndex, readBytes - newlineLength).retain();
```

Comment:
This buffer is shared until closing the ByteBufLineReader. if you want to   keep the sliced buffer, you must copy to new buffer